bodhi/server/models.py

```diff
@@ -2587,6 +2591,20 @@ def url(self):
         url = '/updates/' + self.update.title + '#comment-' + str(self.id)
         return url
 
+    def unique_testcase_feedback(self):
+        """This will filter out duplicates for testcases. It will return the
+        correct number of testcases in testcase_feedbacks.
+        """
+        feedbacks = self.testcase_feedback
+        unique_feedbacks = dict()
+        filtered_feedbacks = list()
+        for feedback in feedbacks:
+            if not unique_feedbacks.get(feedback.testcase.name):
+                unique_feedbacks[feedback.testcase.name] = True
+                filtered_feedbacks.append(feedback)
```

Comment:
It seems like this is using a ```dict``` to achieve what a ```set``` is for. I recommend adding the testcase names to a set to simplify this a bit and to make it clearer what it's doing.